plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/TableStatisticsWriter.java

```diff
@@ -219,28 +213,33 @@ private CollectedStatistics mergeStatisticsIfNecessary(
                             List<BlobMetadata> toRead = reader.fileMetadata().blobs().stream()
                                     .filter(blobMetadata -> blobMetadata.type().equals(APACHE_DATASKETCHES_THETA_V1))
                                     .filter(blobMetadata -> blobMetadata.inputFields().size() == 1)
-                                    .filter(blobMetadata -> pendingPreviousNdvSketches.contains(getOnlyElement(blobMetadata.inputFields())))
                                     .collect(toImmutableList());
                             for (Pair<BlobMetadata, ByteBuffer> read : reader.readAll(toRead)) {
                                 Integer fieldId = getOnlyElement(read.first().inputFields());
-                                checkState(pendingPreviousNdvSketches.remove(fieldId), "Unwanted read of stats for field %s", fieldId);
-                                Memory memory = Memory.wrap(ByteBuffers.getBytes(read.second())); // Memory.wrap(ByteBuffer) results in a different deserialized state
-                                CompactSketch previousSketch = CompactSketch.wrap(memory);
-                                CompactSketch newSketch = requireNonNull(collectedNdvSketches.get(fieldId), "ndvSketches.get(fieldId) is null");
-                                ndvSketches.put(fieldId, SetOperation.builder().buildUnion().union(previousSketch, newSketch));
+                                if (columnsWithRecentlyComputedStats.remove(fieldId)) {
+                                    CompactSketch newSketch = requireNonNull(collectedNdvSketches.get(fieldId), "ndvSketches.get(fieldId) is null");
+                                    newSketch = SetOperation.builder().buildUnion().union(readPreviousSketch(read), newSketch);
+                                    ndvSketches.put(fieldId, newSketch);
+                                }
                             }
                         }
                         catch (IOException exception) {
                             throw new UncheckedIOException(exception);
                         }
+                        columnsWithRecentlyComputedStats.forEach(newFieldId -> ndvSketches.put(newFieldId, collectedNdvSketches.get(newFieldId)));
```

Comment:
This means "for any column we have only partial data (previously had no NDV information), treat this partial data as complete information".
So if i insert 1 row into a table with 1 M rows, i can have NDV of 1 for such column.

Am i reading this right?

Reply:
yes you are right